cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/BusinessMetadataStore.java

```diff
@@ -97,6 +99,21 @@ public Void apply(BusinessMdsIterable input) throws Exception {
   }
 
   /**
+   * @return a {@link MetadataRecord} representing all the metadata (including properties and tags) for the specified
+   * {@link Id.NamespacedId}.
+   */
+  public Set<MetadataRecord> getMetadata(final Id.NamespacedId entityId) {
```

Comment:
Depends on whether we should do both in a single transaction. Is your point that it would be too expensive to club together into one? If so, I agree, I'll make the change you suggested.